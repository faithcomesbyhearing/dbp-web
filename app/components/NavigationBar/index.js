/**
 *
 * NavigationBar
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import dynamic from 'next/dynamic';
import SvgWrapper from '../SvgWrapper';
const TextSelection = dynamic(import('../../containers/TextSelection'), {
	loading: () => null,
});
const ChapterSelection = dynamic(import('../../containers/ChapterSelection'), {
	loading: () => null,
});

class NavigationBar extends React.PureComponent {
	render() {
		const {
			activeTextId,
			activeBookName,
			activeTextName,
			toggleChapterSelection,
			toggleVersionSelection,
			activeChapter,
			isChapterSelectionActive,
			isVersionSelectionActive,
			theme,
			userAgent,
			textDirection,
		} = this.props;
		// may need to wrap each of these in a container div to fix the hover issues
		return (
			<div id={'navigation-bar'} className={'nav-background'}>
				<div className="nav-container">
					<a
						className="logo"
						href={'http://www.bible.is'}
						title={'http://www.bible.is'}
						rel={'noopener'}
					>
						{theme === 'paper' && userAgent !== 'ms' && (
							<svg className={'svg'}>
								<use xlinkHref={'/light_theme_logo.svg#bible.is_logo_light'} />
							</svg>
						)}
						{(theme !== 'paper' || userAgent === 'ms') && (
							<svg className={'svg'} fill={userAgent === 'ms' ? '#fff' : ''}>
								<use xlinkHref={'/dark_theme_logo.svg#bible.is_logo'} />
							</svg>
						)}
					</a>
					<div
						id={'version-dropdown-button'}
						onClick={toggleVersionSelection}
						className="version"
					>
						<SvgWrapper
							className="svg icon"
							fill="#fff"
							svgid="arrow_down"
							opacity=".5"
						/>
						<h1
							title={activeTextName || `${activeTextId} Version`}
							className={'version-text'}
						>
							{activeTextName
								? `${
										activeTextId.slice(0, 3) === 'ENG'
											? `${activeTextId.slice(3)} - `
											: ''
									}${activeTextName}`
								: `${activeTextId} Version`}
						</h1>
						{isVersionSelectionActive && (
							<TextSelection active={isVersionSelectionActive} />
						)}
					</div>
					<div
						id={'chapter-dropdown-button'}
						onClick={toggleChapterSelection}
						className="book-chapter"
					>
						<SvgWrapper
							className="svg icon"
							fill="#fff"
							svgid="arrow_down"
							opacity=".5"
						/>
						<h1
							title={
								activeBookName
									? `${activeBookName} ${activeChapter}`
									: 'No Book Selected'
							}
							className={
								textDirection === 'rtl'
									? 'book-chapter-text rtl'
									: 'book-chapter-text'
							}
						>
							{activeBookName
								? `${activeBookName} ${activeChapter}`
								: 'No Book Selected'}
						</h1>
						{isChapterSelectionActive && (
							<ChapterSelection active={isChapterSelectionActive} />
						)}
					</div>
				</div>
			</div>
		);
	}
}

NavigationBar.propTypes = {
	theme: PropTypes.string,
	userAgent: PropTypes.string,
	activeTextId: PropTypes.string,
	activeBookName: PropTypes.string,
	activeTextName: PropTypes.string,
	textDirection: PropTypes.string,
	toggleChapterSelection: PropTypes.func,
	toggleVersionSelection: PropTypes.func,
	activeChapter: PropTypes.number,
	isChapterSelectionActive: PropTypes.bool,
	isVersionSelectionActive: PropTypes.bool,
};

export default NavigationBar;
