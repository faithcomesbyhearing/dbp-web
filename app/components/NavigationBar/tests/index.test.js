import React from 'react';
import renderer from 'react-test-renderer';
import NavigationBar from '..';

const props = {
	theme: 'red',
	userAgent: '',
	activeTextId: 'ENGESV',
	activeBookName: 'Matthew',
	activeTextName: 'English Standard Version',
	textDirection: 'ltr',
	toggleChapterSelection: jest.fn(),
	toggleVersionSelection: jest.fn(),
	activeChapter: 1,
	isChapterSelectionActive: false,
	isVersionSelectionActive: false,
};

describe('<NavigationBar /> component', () => {
	it('should match snapshot with default props', () => {
		const tree = renderer.create(<NavigationBar {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match snapshot with chapter selection active', () => {
		const tree = renderer
			.create(<NavigationBar {...props} isChapterSelectionActive />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match snapshot with version selection active', () => {
		const tree = renderer
			.create(<NavigationBar {...props} isVersionSelectionActive />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match snapshot with ms as the user agent', () => {
		const tree = renderer
			.create(<NavigationBar {...props} userAgent={'ms'} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match snapshot with paper theme', () => {
		const tree = renderer
			.create(<NavigationBar {...props} theme={'paper'} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should match snapshot with paper theme and ms as the user agent', () => {
		const tree = renderer
			.create(<NavigationBar {...props} theme={'paper'} userAgent={'ms'} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
