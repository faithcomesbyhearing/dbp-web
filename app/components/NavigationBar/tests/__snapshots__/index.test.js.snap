// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> component should match snapshot with chapter selection active 1`] = `
<div
  className="nav-background"
  id="navigation-bar"
>
  <div
    className="nav-container"
  >
    <a
      className="logo"
      href="http://www.bible.is"
      rel="noopener"
      title="http://www.bible.is"
    >
      <svg
        className="svg"
        fill=""
      >
        <use
          xlinkHref="/dark_theme_logo.svg#bible.is_logo"
        />
      </svg>
    </a>
    <div
      className="version"
      id="version-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="version-text"
        title="English Standard Version"
      >
        ESV - English Standard Version
      </h1>
    </div>
    <div
      className="book-chapter"
      id="chapter-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="book-chapter-text"
        title="Matthew 1"
      >
        Matthew 1
      </h1>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> component should match snapshot with default props 1`] = `
<div
  className="nav-background"
  id="navigation-bar"
>
  <div
    className="nav-container"
  >
    <a
      className="logo"
      href="http://www.bible.is"
      rel="noopener"
      title="http://www.bible.is"
    >
      <svg
        className="svg"
        fill=""
      >
        <use
          xlinkHref="/dark_theme_logo.svg#bible.is_logo"
        />
      </svg>
    </a>
    <div
      className="version"
      id="version-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="version-text"
        title="English Standard Version"
      >
        ESV - English Standard Version
      </h1>
    </div>
    <div
      className="book-chapter"
      id="chapter-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="book-chapter-text"
        title="Matthew 1"
      >
        Matthew 1
      </h1>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> component should match snapshot with ms as the user agent 1`] = `
<div
  className="nav-background"
  id="navigation-bar"
>
  <div
    className="nav-container"
  >
    <a
      className="logo"
      href="http://www.bible.is"
      rel="noopener"
      title="http://www.bible.is"
    >
      <svg
        className="svg"
        fill="#fff"
      >
        <use
          xlinkHref="/dark_theme_logo.svg#bible.is_logo"
        />
      </svg>
    </a>
    <div
      className="version"
      id="version-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="version-text"
        title="English Standard Version"
      >
        ESV - English Standard Version
      </h1>
    </div>
    <div
      className="book-chapter"
      id="chapter-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="book-chapter-text"
        title="Matthew 1"
      >
        Matthew 1
      </h1>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> component should match snapshot with paper theme 1`] = `
<div
  className="nav-background"
  id="navigation-bar"
>
  <div
    className="nav-container"
  >
    <a
      className="logo"
      href="http://www.bible.is"
      rel="noopener"
      title="http://www.bible.is"
    >
      <svg
        className="svg"
      >
        <use
          xlinkHref="/light_theme_logo.svg#bible.is_logo_light"
        />
      </svg>
    </a>
    <div
      className="version"
      id="version-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="version-text"
        title="English Standard Version"
      >
        ESV - English Standard Version
      </h1>
    </div>
    <div
      className="book-chapter"
      id="chapter-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="book-chapter-text"
        title="Matthew 1"
      >
        Matthew 1
      </h1>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> component should match snapshot with paper theme and ms as the user agent 1`] = `
<div
  className="nav-background"
  id="navigation-bar"
>
  <div
    className="nav-container"
  >
    <a
      className="logo"
      href="http://www.bible.is"
      rel="noopener"
      title="http://www.bible.is"
    >
      <svg
        className="svg"
        fill="#fff"
      >
        <use
          xlinkHref="/dark_theme_logo.svg#bible.is_logo"
        />
      </svg>
    </a>
    <div
      className="version"
      id="version-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="version-text"
        title="English Standard Version"
      >
        ESV - English Standard Version
      </h1>
    </div>
    <div
      className="book-chapter"
      id="chapter-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="book-chapter-text"
        title="Matthew 1"
      >
        Matthew 1
      </h1>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> component should match snapshot with version selection active 1`] = `
<div
  className="nav-background"
  id="navigation-bar"
>
  <div
    className="nav-container"
  >
    <a
      className="logo"
      href="http://www.bible.is"
      rel="noopener"
      title="http://www.bible.is"
    >
      <svg
        className="svg"
        fill=""
      >
        <use
          xlinkHref="/dark_theme_logo.svg#bible.is_logo"
        />
      </svg>
    </a>
    <div
      className="version"
      id="version-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="version-text"
        title="English Standard Version"
      >
        ESV - English Standard Version
      </h1>
    </div>
    <div
      className="book-chapter"
      id="chapter-dropdown-button"
      onClick={[MockFunction]}
    >
      <svg
        className="svg icon"
        fill="#fff"
        opacity=".5"
      >
        <use
          xlinkHref="/svglist.svg#arrow_down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <h1
        className="book-chapter-text"
        title="Matthew 1"
      >
        Matthew 1
      </h1>
    </div>
  </div>
</div>
`;
